roles/postgres/tasks/check_postgres_version.yml

```diff
@@ -0,0 +1,76 @@
+---
+
+# It is possible that N-2 postgres pods may still be present in the namespace from previous upgrades.
+# So we have to take that into account and preferentially set the most recent one.
+- name: Get the old postgres pod (N-1)
+  k8s_info:
+    kind: Pod
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    field_selectors:
+      - status.phase=Running
```

Comment:
Instead of getting old pods running in the namespace and then filter in next task, can't we just get the postgres pod by adding a field selector? There must be a label we can use?

Reply:
We can't know for sure if the old postgres pod is version 12 or 13. That's why I did it this way.